projects/native-ephemery-client-pair-implementation.md

```diff
@@ -0,0 +1,148 @@
+
+The primary challenge is managing the dynamic genesis, which resets periodically, unlike other networks with hardcoded genesis. 
+
+Integrating Ephemery into clients like Teku and Besu enhances its usability by allowing configuration via a simple flag. This integration promotes experimentation and validation, strengthening the Ethereum ecosystem. 
+
+The project focuses on client configuration and genesis handling, aiming to streamline testnet setup and operation.
+
+## Project Description
+The proposed solution involves implementing native support for the Ephemery testnet in both Teku (an Ethereum consensus client) and Besu (an Ethereum execution client). This will be achieved by introducing a new --ephemery flag that handles the dynamic genesis requirement. 
+
+**The solution includes:**
+- Developing a mechanism within each client to calculate or download the new genesis periodically.
+- Creating an "Ephemery lib" or submodule within each client to handle genesis generation and reset functionality.
+- Ensuring seamless integration with existing client architecture to support Ephemery's unique requirements.
+
+## Specification
+
+The detailed spec can be be found here [eip6916](https://eips.ethereum.org/EIPS/eip-6916)
+
+**Feasibility Study:**
+
+- Assess current client architectures and identify potential integration points for Ephemery.
+- We conducted detailed technical analyses of clients, especially focusing on the reset function.
+- Compile a feasibility report with findings and recommendations.
+
+**Genesis Functionality:**
+
+- Implement a dynamic genesis handling mechanism within both Teku and Besu clients.
+- Develop a library/submodule in Java to manage genesis calculation or retrieval.
+
+**Client Integration:**
+
+- Modify the initialization process in both clients to recognize the --ephemery flag.
+- Ensure that the clients can fetch or compute the genesis block before starting.
+
+**Documentation and Tutorials:**
+- Create comprehensive documentation and tutorials for developers to understand and utilize the new Ephemery functionality.
+
+**Website Development:**
+- Build an Ephemery website featuring a reset countdown, project overview, and documentation.
+
+**Technical Details:**
+
+**Teku Modifications:**
+- Integrate the Ephemery lib to handle consensus layer genesis.
+- Ensure compatibility with Teku's existing architecture for seamless integration.
+
+**Besu Modifications:**
+- Integrate the Ephemery lib for execution layer genesis handling.
+- Align with Besu's architecture to maintain performance and stability.
+
+## Roadmap
+
+**Initial Research and Planning (2 weeks):**
+- Understand client architectures and feasibility.
+- Draft detailed implementation plans.
+
+**Library/Submodule Development (4 weeks):**
+- Develop the Ephemery lib for genesis handling.
+
+**Client Integration (6 weeks):**
```

Comment:
The timing is a bit weird here. Adding a flag itself is trivial, the important part is what it actually triggers.